### PR TITLE
add build pod name annotation to builds

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/5692

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal. @jwforres fyi and to confirm you're happy w/ the annotation name.

**Comment by bparees** [comment]
[test][extended:core]

**Comment by jwforres** [comment]
Annotation name lgtm

**Comment by smarterclayton** [comment]
Annotation name is fine with me as well.

**Comment by csrwng** [comment]
LGTM

**Comment by mfojtik** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

**Comment by liggitt** [comment]
```
E1112 13:56:47.944289   15343 config_controller.go:68] error instantiating Build from BuildConfig /testBuildConfig: error
--- FAIL: TestHandleBuild (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
```

File: pkg/build/controller/controller.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 		build.Status.Reason = buildapi.StatusReasonCannotCreateBuildPod
 		return fmt.Errorf("failed to create build pod: %v", err)
 	}
-
+	build.Annotations[buildapi.BuildPodNameAnnotation] = podSpec.Name
```
Comment (liggitt) [inline pkg/build/controller/controller.go]: possible for `build.Annotations` to be nil?

Comment (bparees) [inline pkg/build/controller/controller.go]: news to me, i would have expected it to always get initialized deeper in the api, but i guess i can initialize it.

Comment (bparees) [inline pkg/build/controller/controller.go]: looks like the normal build creation flow always populates annotations, but i'm guessing in some test scenarios we don't go through that flow.  check/initialization added.

**Comment by bparees** [comment]
[merge]

**Pull request merged**